.eslintrc.js

```diff
@@ -15,55 +15,51 @@ module.exports = {
     },
   },
   rules: {
-    'import/prefer-default-export': 'off',
     'simple-import-sort/imports': 'warn',
     quotes: [2, 'single', { avoidEscape: true }],
     'prettier/prettier': ['error', { singleQuote: true }],
 
     // TODO: ticket all of these to be enabled or explain why they are disabled
+    // Destructuring props is not required because it clutters the function type signature.
+    // Props can be destructured in a separate line.
     'react/destructuring-assignment': 'off',
+    // Use TypeScript instead of PropTypes for new code.
     'react/forbid-prop-types': 'off',
-    'react/hooks/exhaustive-deps': 'off',
     'react/prop-types': 'off',
     'react/require-default-props': 'off',
     'react/sort-comp': 'off',
     'react/static-property-placement': 'off',
 
-    'import/extensions': 'off',
     'import/named': 'off',
     'import/no-cycle': 'off',
-    'import/no-extraneous-dependencies': 'off',
     'import/no-named-as-default': 'off',
-    'import/no-named-as-default-member': 'off',
-    'import/no-relative-parent-imports': 'off',
+    // Named exports are more convenient for mocking.
+    'import/prefer-default-export': 'off',
 
+    // TS directive comments will be necessary during transition from JS to TS.
     '@typescript-eslint/ban-ts-comment': 'off',
     '@typescript-eslint/ban-types': 'off',
-    '@typescript-eslint/default-param-last': 'off',
-    '@typescript-eslint/naming-convention': 'off',
     '@typescript-eslint/no-empty-function': 'off',
-    '@typescript-eslint/no-explicit-any': ['off'],
-    '@typescript-eslint/no-implied-eval': 'off',
+    // `any` is useful for incremental type improvements during the transition from JS to TS.
+    '@typescript-eslint/no-explicit-any': 'off',
     '@typescript-eslint/no-non-null-assertion': 'off',
     '@typescript-eslint/no-shadow': 'off',
     '@typescript-eslint/no-throw-literal': 'off',
-    '@typescript-eslint/no-unused-expressions': 'off',
     '@typescript-eslint/no-use-before-define': 'off',
-    '@typescript-eslint/no-var-requires': 'off',
     '@typescript-eslint/return-await': 'off',
     '@typescript-eslint/strict-boolean-expressions': 'off', // TODO: should be 'warn',
 
     'class-methods-use-this': 'off',
     'consistent-return': 'off',
     'default-param-last': 'off',
     'func-names': 'off',
-    'implicit-arrow-linebreak': 'off',
-    'max-len': 'off', // TODO: should be: ['warn', { code: 120 }],
+    // Formatting is handled by Prettier.
```

Comment:
This configuration is incompatible with Prettier.

We configure Prettier's [printWidth](https://prettier.io/docs/en/options.html#print-width) to 120 via `.editorconfig`. Prettier may generate lines longer than `printWidth`, so this will conflict with the Prettier rule. We could configure `max-len` to be larger than `printWidth` to reduce the probability of a conflict.

Reply:
I wouldn't mind the warning length being 130 or thereabouts, it'd just be nice to get a squiggly red/yellow/blue line when you go over the line limit

Reply:
seems like this topic is related to the recent PR to add prettierrc file...

Reply:
Still though, Prettier doesn't guarantee lines generated are less than `printWidth + 10`, so while that might reduce the chance of a conflict, it's still there.

Reply:
I'd just like some sort of nudge in the right direction. We can't hold people to the standard of always remembering to keep their lines shorter than 1xx in length, any automated mechanism that at least encourages mindfulness of this should be sought

Reply:
As someone who has lived in repos with line length limits, I very much don't care about it. If you want to keep your own comments at a particular length, most IDEs will allow you to configure a line length indication.

Reply:
TIL `max-len` supports different options for code and comments. So if desired we _could_ set code to something like 1000 that won't conflict with Prettier and set comments to 120. That said, I don't really think we need to _enforce_ a line length for comments.